File: ferry_booking_system/app/Models/VehicleCategory.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class VehicleCategory extends Model
{
    use HasFactory;

    /**
     * The attributes that are mass assignable.
     *
     * @var array<int, string>
     */
    protected $fillable = [
        'code',
        'name',
        'description',
        'vehicle_type',
        'base_price',
        'is_active',
    ];

    /**
     * The attributes that should be cast.
     *
     * @var array<string, string>
     */
    protected $casts = [
        'base_price' => 'decimal:2',
        'is_active' => 'boolean',
    ];

    /**
     * Get the vehicles for this category.
     */
    public function vehicles()
    {
        return $this->hasMany(Vehicle::class);
    }

    /**
     * Format the price as rupiah
     */
    public function getFormattedPriceAttribute()
    {
        return 'Rp ' . number_format($this->base_price, 0, ',', '.');
    }

    /**
     * Scope a query to only include active categories.
     */
    public function scopeActive($query)
    {
        return $query->where('is_active', true);
    }

    /**
     * Scope a query to filter by vehicle type.
     */
    public function scopeOfType($query, $type)
    {
        return $query->where('vehicle_type', $type);
    }
}

File: ferry_booking_app/lib/models/vehicle_category.dart
class VehicleCategory {
  final int id;
  final String code;
  final String name;
  final String vehicleType;
  final double basePrice;
  final bool isActive;
  final String? description;

  VehicleCategory({
    required this.id,
    required this.code,
    required this.name,
    required this.vehicleType,
    required this.basePrice,
    required this.isActive,
    this.description,
  });

  factory VehicleCategory.fromJson(Map<String, dynamic> json) {
    return VehicleCategory(
      id: int.parse(json['id'].toString()),
      code: json['code'],
      name: json['name'],
      vehicleType: json['vehicle_type'],
      basePrice: double.parse(json['base_price'].toString()),
      isActive: json['is_active'] == true || json['is_active'] == 1,
      description: json['description'],
    );
  }
}